lib/openstax/aws/stack.rb

```diff
@@ -1,5 +1,78 @@
+require 'json'
+
 module OpenStax::Aws
   class Stack
+    class Event
+      def initialize(aws_stack_event)
+        @aws_stack_event = aws_stack_event
+      end
+            
+      def status
+        @aws_stack_event.data.resource_status
+      end
+            
+      def reason
+        @aws_stack_event.data.resource_status_reason
+      end
+            
+      def failure?
+        %w(
+          ROLLBACK_COMPLETE
+          ROLLBACK_IN_PROGRESS
+          CREATE_FAILED
+          ROLLBACK_FAILED
+          DELETE_FAILED
+          UPDATE_ROLLBACK_FAILED
+          IMPORT_ROLLBACK_FAILED
+        ).include?(status)
+      end
+            
+      def user_initiated?
+        reason == "User Initiated"
+      end
+    end
+
+    class Status
+      def initialize(stack)
+        @stack = stack
+      end
+
+      def status_text
+        @stack.aws_stack.stack_status
+      end
+
+      def failure?
+        %w(
+          ROLLBACK_COMPLETE
+          ROLLBACK_IN_PROGRESS
+          CREATE_FAILED
+          ROLLBACK_FAILED
+          DELETE_FAILED
+          UPDATE_ROLLBACK_FAILED
+          IMPORT_ROLLBACK_FAILED
+        ).include?(status_text)
+      end
+
+      def failed_events_since_last_user_event
+        @stack.events.each_with_object([]) do |aws_stack_event, array|
+          event = Event.new(aws_stack_event)
```

Comment:
Something needs to change on line 57 & 58 what is it?

Reply:
I thought that by returning Event objects in the `.event` call I could just skip line 58 but it didn't work

Reply:
it should.  also `aws_stack_event` is not the right var name